Source/RageCommunity.Library/Graphics/Checkpoint.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Drawing;
+using Rage;
+using Rage.Native;
+using RageCommunity.Library.Extensions;
+
+namespace RageCommunity.Library.Graphics
+{
+    /// <summary>
+    /// This class represents a checkpoint within GTA V.
+    /// </summary>
+    /// <remarks>
+    /// Originally created by PNWParksFan. 
+    /// </remarks>
+    public class Checkpoint : IDeletable, ISpatial
+    {
+        private readonly Boolean setToGround;
+        private Color color;
+        private Single height;
+        private Vector3 nextPosition;
+        private Vector3 position;
+
+        public Checkpoint(Vector3 position,
+                          Color color,
+                          Single radius = 1f,
+                          Single height = 1f,
+                          Int32 checkpointType = 47,
+                          Int32 reserved = 0,
+                          Boolean setZtoGround = true) : this(position, position, color, radius, height, checkpointType, reserved, setZtoGround)
+        {
+        }
+
+        public Checkpoint(Vector3 position,
+                          Vector3 nextPosition,
+                          Color color,
+                          Single radius,
+                          Single height,
+                          Int32 checkpointType = 0,
+                          Int32 reserved = 0,
+                          Boolean setZtoGround = true)
+        {
+            CheckpointType = checkpointType;
+            this.position = position;
```

Comment:
If we use the underscore preface, we won't have to reference everything with 'this'

Reply:
Take a read: https://softwareengineering.stackexchange.com/questions/70223/what-is-the-regarded-current-best-practises-regarding-the-this-keyword-in-fron 

using ".this" is the logical consequence based on the guideline to not use prefixes. I would like to stick again with the Microsoft best practice.

Reply:
See other comment about the convention.  We can use that for the discussion.  Resolving this.